src/tests/math_func.cpp

```diff
@@ -0,0 +1,77 @@
+/*
+ * This file is part of OpenTTD.
+ * OpenTTD is free software; you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation, version 2.
+ * OpenTTD is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details. You should have received a copy of the GNU General Public License along with OpenTTD. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/** @file math_func_test.cpp Test functionality from core/math_func. */
+
+#include "../stdafx.h"
+
+#include "../3rdparty/catch2/catch.hpp"
+
+#include "../core/math_func.hpp"
+
+TEST_CASE("LeastCommonMultipleTest - Zero")
+{
+	CHECK(0 == LeastCommonMultiple(0, 0));
+	CHECK(0 == LeastCommonMultiple(0, 600));
+	CHECK(0 == LeastCommonMultiple(600, 0));
+}
+
+TEST_CASE("LeastCommonMultipleTest - FindLCM")
+{
+	CHECK(25 == LeastCommonMultiple(5, 25));
+	CHECK(25 == LeastCommonMultiple(25, 5));
+	CHECK(130 == LeastCommonMultiple(5, 26));
+	CHECK(130 == LeastCommonMultiple(26, 5));
+}
+
+TEST_CASE("GreatestCommonDivisorTest - Negative")
+{
+	CHECK(4 == GreatestCommonDivisor(4, -52));
+	// CHECK(3 == GreatestCommonDivisor(-27, 6)); // error - returns -3
```

Comment:
Commented code, and an error elsewhere?

Reply:
I reckon an error elsewhere. It's a bit via-via-via (#9038 -> #7796 -> #7254), but there was some discussion about it before without real remedy https://github.com/OpenTTD/OpenTTD/pull/7254/files#r259590007
In any case, I copied this test straight from #9038, which copied it from #7796.